pandas/io/sql.py

```diff
@@ -381,7 +381,8 @@ def read_sql(sql, con, index_col=None, coerce_float=True, params=None,
 
     try:
         _is_table_name = pandas_sql.has_table(sql)
-    except (ImportError, AttributeError):
+    except Exception:
```

Comment:
is there a test for this?

Reply:
As I said above, see the issue #24988. I can't get a reproducible case on travis for now

Reply:
ok, if one of the OP can provide a test before release would be good. otherwise ok.